source/line.h

```diff
@@ -1,59 +1,60 @@
 #pragma once
 #include "common.h"
+#include "point.h"
+#include "constants_geom.h"
 
+#define TTT template<typename T>
```

Comment:
I might be a bit late to the conversation, but I really don't like hiding templates away like this. It makes the resulting `T` type obscure in the code that uses it. IMO, we should use `ld` in code that requires it (like `Line`), and only take points of that type.

If we allow passing integer coordinate points we hide away the cast, and might trick the user into thinking they are using code that maintains accuracy, while loosing it with doubles. Would enforcing a call to `to_ld` be detrimental in any way? It looks like the best solution to me, it doesn't hide precision loss and would also remove the need for templating